java-engine/src/main/java/io/getunleash/engine/CustomStrategiesEvaluator.java

```diff
@@ -0,0 +1,190 @@
+package io.getunleash.engine;
+
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class CustomStrategiesEvaluator {
+  private static final Logger log = LoggerFactory.getLogger(CustomStrategiesEvaluator.class);
+  static final String EMPTY_STRATEGY_RESULTS = "{}";
+  private final Map<String, IStrategy> registeredStrategies;
+  private final Set<String> builtinStrategies;
+
+  private final IStrategy fallbackStrategy;
+  private final ObjectMapper mapper;
+
+  private Map<String, List<MappedStrategy>> featureStrategies = new HashMap<>();
+
+  public CustomStrategiesEvaluator(
+      Stream<IStrategy> customStrategies, Set<String> builtinStrategies) {
+    this(customStrategies, null, builtinStrategies);
+  }
+
+  public CustomStrategiesEvaluator(
+      Stream<IStrategy> customStrategies,
+      IStrategy fallbackStrategy,
+      Set<String> builtinStrategies) {
+    this.builtinStrategies = builtinStrategies;
+    this.mapper = new ObjectMapper();
+    this.mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    this.registeredStrategies =
+        customStrategies.collect(toMap(IStrategy::getName, identity(), (a, b) -> a));
+    this.fallbackStrategy = fallbackStrategy;
+  }
+
+  public void loadStrategiesFor(String toggles) {
+    if (this.registeredStrategies.isEmpty() && this.fallbackStrategy == null) {
+      return;
+    }
+
+    if (toggles == null || toggles.isEmpty()) {
+      return;
+    }
+
+    try {
+      VersionedFeatures wrapper =
```

Comment:
What's the "versioned" part referring to?

Reply:
That's a good question and I actually don't know. I didn't write that, just uncommenting it

Reply:
Sorry, undeleting it